### ISSUE TITLE
feature enhancement: add a new field orderQuantity to OrderEvent, and a new method amount()

### ISSUE THREAD

**Issue by mingwugmail** [QuantConnect/Lean#4237]
#### Expected Behavior

Currently, OrderEvent only has fillQuantity defined. It will be more informative to add orderQuantity.

https://github.com/QuantConnect/Lean/blob/cd08b8c16c78fdc5c702dca6c739c1a78bcbaa6d/Common/Orders/OrderEvent.cs#L28

e.g. for a confirmed Submitted OrderEvent: 

https://groups.google.com/forum/#!topic/lean-engine/u5VEO_-OBw8

I enter a position using this function:
```
  SetHoldings(_symbol, 1.0);
```
instead of create my own order.

Then I listen on the callback:
```
        public override void OnOrderEvent(OrderEvent orderEvent) {
             Debug($"OnOrderEvent: {orderEvent}");
        }
```
The 1st message I got is:

2020-03-12 12:00:00 OnOrderEvent: Time: 03/12/2020 16:00:00 OrderID: 1 Symbol: MSFT Status: Submitted

But this does not have the Quantity info.It will be hard to track this submitted orders later on, esp. in case of multiple partial fills.

I'm wondering how can I get these detail info of the Submitted orders.

Same question for:
```
  Liquidate(_symbol);
```
(using Portfolio[_symbol].Quantity is not quite right, I want that status of the confirmed "Submitted" order's Quantity; e.g. just suppose there are some other new issues such as float rounding in the code, that Portfolio[_symbol].Quantity != submittedOrder.Quantity).

#### Potential Solution

add a new field orderQuantity to OrderEvent, and populate it properly.

Now a Submitted OrderEvent will look like, e.g.:
```
orderEvent.orderQuantity = 100;
orderEvent.fillQuantity = 0;
```

and a partially filled OrderEvent object will look like, e.g.:
```
orderEvent.orderQuantity = 100;
orderEvent.fillQuantity = 66;
```

**Comment by mingwugmail** [comment]
Also, can we also add a method: amount()?
```
public decimal amount() {
  return fillQuantity * fillPrice;
}
```

it will be much more convenient to use this object.

**Comment by Martin-Molinero** [comment]
Hi @mingwugmail, at https://github.com/QuantConnect/Lean/pull/4159 we have added the Quantity field as well as StopPrice and LimitPrice for order events. We also added a new OrderStatus.UpdateSubmitted status. 
> Also, can we also add a method: amount()?

The value of the order depends on the type of Order. Currently, in Lean master the Order has a `GetValue(Security)` method which acts accordingly for each type.
